### PR TITLE
[ISSUE#12359]Refactoring the historical configuration cleanup strategy with SPI 

### PR DESCRIPTION
**Pull request by Sunrisea**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

fix issue#12359

## Brief changelog

Refactoring the historical configuration cleanup strategy with SPI (Service Provider Interface), and add unit test for
DefaultHistoryConfigCleaner ,HistoryConfigCleanerManager and HistoryConfigCleanerConfig
## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by KomachiSion — COMMENTED** [review]
这种方案是否要求拓展者必须依赖nacos-config模块才可以开发拓展？

**Comment by Sunrisea** [comment]
> 这种方案是否要求拓展者必须依赖nacos-config模块才可以开发拓展？

是的

File: config/src/test/java/com/alibaba/nacos/config/server/service/dump/DefaultHistoryConfigCleanerTest.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        Mockito.verify(historyConfigInfoPersistService, Mockito.times(0))
+                .removeConfigHistory(any(Timestamp.class), anyInt());
+        
+        DatasourceConfiguration.setEmbeddedStorage(true);
+        envUtilMockedStatic.when(EnvUtil::getStandaloneMode).thenReturn(true);
+        EnvUtil.setIsStandalone(true);
+        defaultHistoryConfigCleaner.cleanConfigHistory();
+        Mockito.verify(historyConfigInfoPersistService, Mockito.times(1))
+                .removeConfigHistory(any(Timestamp.class), anyInt());
+        envUtilMockedStatic.close();
```
Comment (shiyiyue1102) [inline config/src/test/java/com/alibaba/nacos/config/server/service/dump/DefaultHistoryConfigCleanerTest.java:162]: 如果这个用例执行报错，这里就无法关闭 mock static，会导致其他用例失败

File: config/src/main/java/com/alibaba/nacos/config/server/service/dump/HistoryConfigCleaner.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    
+    /**
+     * Clean history config.
+     */
+    public void cleanHistoryConfig();
+    
+    /**
+     * Start clean task.
+     */
+    public void startCleanTask();
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/service/dump/HistoryConfigCleaner.java:33]: move schedule task to  dumpservice

File: config/src/main/java/com/alibaba/nacos/config/server/service/dump/DumpService.java [hunk 3]
```diff
@@ -194,29 +190,23 @@ void handleConfigDataChange(Event event) {
...
+        private HistoryConfigCleaner historyConfigCleaner;
+        
+        public ConfigHistoryClear(HistoryConfigCleaner historyConfigCleaner) {
+            this.historyConfigCleaner = historyConfigCleaner;
+        }
+        
         @Override
         public void run() {
-            clearConfigHistory();
+            LOGGER.warn("clearHistoryConfig start");
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/service/dump/DumpService.java:206]: 1.when canExecute return false/true ,print log also ,like 'history config cleaner is not enable in current context' , 'history config cleaner is enable in current context, trying to run cleaner.'
2.when cleaner run finish ,print log like 'history cleaner finished successfully' and should catch exception when  cleaner throw exception ,or it wont be scheduled next period.

**Review by shiyiyue1102 — APPROVED** [review]

**Pull request merged**